includes/models/model.llms.student.php

```diff
@@ -1435,6 +1462,73 @@ public function unenroll( $product_id, $trigger = 'any', $new_status = 'expired'
 
 	}
 
+	/**
+	 * Delete a student enrollment.
+	 *
+	 * @since [version]
+	 *
+	 * @param  int     $product_id  WP Post ID of the course or membership.
+	 * @param  string  $trigger     Optiona. Only delete the student's enrollment if the original enrollment trigger matches the submitted value
+	 *                              "any" will remove regardless of enrollment trigger. Default "any".
+	 * @return boolean Whether or not the enrollment records have been succesfully removed.
+	 *
+	 * @see  llms_delete_student_enrollment() calls this function without having to instantiate the LLMS_Student class first.
+	 */
+	public function delete_enrollment( $product_id, $trigger = 'any' ) {
+
+		// assume we can't delete the enrollment
+		$delete = false;
+
+		// if trigger is "any" we'll just delete the enrollment regardless of the trigger.
+		if ( 'any' === $trigger ) {
+
+			$delete = true;
+
+		} // End if().
+		else {
+
+			$enrollment_trigger = $this->get_enrollment_trigger( $product_id );
+
+			// no enrollment trigger exists b/c pre 3.0.0 enrollment, delete the user's enrollment.
+			if ( ! $enrollment_trigger ) {
+
+				$delete = apply_filters( 'lifterlms_legacy_delete_enrollment_action', true );
+
+			} // End if().
+			elseif ( $enrollment_trigger === $trigger ) {
+
+				$delete = true;
+
+			}
+		}
+
+		// delete if we can
+		if ( $delete ) {
+
+			// Unenroll the student if enrolled
+			$this->unenroll( $product_id, $trigger, 'expired' );
```

Comment:
I don't think we need to unenroll before deleting. At this point they should already be unenrolled based on the interface, however, if someone wants to programmatically delete without running those checks that's just fine by me, I think.

I didn't run this but I think if this line is removed it just cuts down on the number of queries that need to be executed.

If I'm reading this all properly we should be able to simply delete the records without unenrolling first.

Reply:
I thought the same at the beginning, also to have this method doing one and one thing only.
But then I thought that in the rare case someone wanted to use this public method it should trigger the unenroll as well.

But yes I'm going to remove this line and amend the documentation here:
https://github.com/gocodebox/lifterlms/pull/843/files/4c7bbab3ed0717eeb3fc0fba2d01a67ea72eb097#diff-337dc0e7959977273f0d8630e90dd68fR403
replacing it with something that says the opposite, like "this method doesn't unenroll the student, you might want to run llms_unenroll_student before"

Reply:
I've changed idea about the documentation "replacement", I don't think it's needed.
Fixed in 3e0ced7